--- a/lua/lspconfig/configs/volar.lua	1970-01-01 07:00:01.000000000 +0700
+++ b/lua/lspconfig/configs/volar.lua	2025-01-26 17:15:09.408529914 +0700
@@ -1,35 +1,35 @@
-local util = require 'lspconfig.util'
+local util = require("lspconfig.util")

 local function get_typescript_server_path(root_dir)
-  local project_root = vim.fs.find('node_modules', { path = root_dir, upward = true })[1]
-  return project_root and (project_root .. '/typescript/lib') or ''
+	local project_root = vim.fs.dirname(vim.fs.find("node_modules", { path = root_dir, upward = true })[1])
+	return project_root and (project_root .. "/node_modules/typescript/lib") or ""
 end

 -- https://github.com/vuejs/language-tools/blob/master/packages/language-server/lib/types.ts
 local volar_init_options = {
-  typescript = {
-    tsdk = '',
-  },
+	typescript = {
+		tsdk = "",
+	},
 }

 return {
-  default_config = {
-    cmd = { 'vue-language-server', '--stdio' },
-    filetypes = { 'vue' },
-    root_dir = util.root_pattern 'package.json',
-    init_options = volar_init_options,
-    on_new_config = function(new_config, new_root_dir)
-      if
-        new_config.init_options
-        and new_config.init_options.typescript
-        and new_config.init_options.typescript.tsdk == ''
-      then
-        new_config.init_options.typescript.tsdk = get_typescript_server_path(new_root_dir)
-      end
-    end,
-  },
-  docs = {
-    description = [[
+	default_config = {
+		cmd = { "vue-language-server", "--stdio" },
+		filetypes = { "vue" },
+		root_dir = util.root_pattern("package.json"),
+		init_options = volar_init_options,
+		on_new_config = function(new_config, new_root_dir)
+			if
+				new_config.init_options
+				and new_config.init_options.typescript
+				and new_config.init_options.typescript.tsdk == ""
+			then
+				new_config.init_options.typescript.tsdk = get_typescript_server_path(new_root_dir)
+			end
+		end,
+	},
+	docs = {
+		description = [[
 https://github.com/vuejs/language-tools/tree/master/packages/language-server

 Volar language server for Vue
@@ -103,5 +103,5 @@
 }
 ```
     ]],
-  },
+	},
 }
