--- lua/lspconfig/configs/astro.lua	2025-01-25 23:45:05.481132375 +0700
+++ lua/lspconfig/configs/astro.lua	2025-01-25 23:45:24.481126498 +0700
@@ -2,7 +2,7 @@

 local function get_typescript_server_path(root_dir)
 	local project_root = vim.fs.dirname(vim.fs.find("node_modules", { path = root_dir, upward = true })[1])
-	return project_root and (project_root .. "/typescript/lib") or ""
+	return project_root and (project_root .. "/node_modules/typescript/lib") or ""
 end

 return {
